OpenRA.Game/Map/MapCache.cs

```diff
@@ -241,39 +242,60 @@ public void QueryRemoteMapDetails(string repositoryUrl, IEnumerable<string> uids
 			{
 				var client = HttpClientFactory.Create();
 				var stringPool = new HashSet<string>(); // Reuse common strings in YAML
+				var buffer = new StringBuilder();
 
 				// Limit each query to 50 maps at a time to avoid request size limits
 				for (var i = 0; i < queryUids.Count; i += 50)
 				{
 					var batchUids = queryUids.Skip(i).Take(50).ToList();
 					var url = repositoryUrl + "hash/" + string.Join(",", batchUids) + "/yaml";
-					try
-					{
-						var httpResponseMessage = await client.GetAsync(url);
-						var result = await httpResponseMessage.Content.ReadAsStreamAsync();
-
-						var yaml = MiniYaml.FromStream(result, url, stringPool: stringPool);
-						foreach (var kv in yaml)
-							previews[kv.Key].UpdateRemoteSearch(MapStatus.DownloadAvailable, kv.Value, mapDetailsReceived);
 
-						foreach (var uid in batchUids)
-						{
-							var p = previews[uid];
-							if (p.Status != MapStatus.DownloadAvailable)
-								p.UpdateRemoteSearch(MapStatus.Unavailable, null, null);
-						}
-					}
-					catch (Exception e)
+					using (new PerfTimer("RemoteMapDetails"))
 					{
-						Log.Write("debug", "Remote map query failed with error:");
-						Log.Write("debug", e);
-						Log.Write("debug", $"URL was: {url}");
+						try
+						{
+							await using (var resultStream = await client.GetStreamAsync(url))
+							{
+								using (var resultReader = new StreamReader(resultStream))
+								{
+									while (true)
+									{
+										var line = await resultReader.ReadLineAsync();
+										if (line == null || !line.StartsWith('\t'))
+										{
+											var yaml = MiniYaml.FromString(buffer.ToString(), url, stringPool: stringPool);
+											buffer.Clear();
+											foreach (var kv in yaml)
+												previews[kv.Key].UpdateRemoteSearch(MapStatus.DownloadAvailable, kv.Value, mapDetailsReceived);
+
+											if (line == null)
+												break;
+										}
+
+										buffer.Append(line);
```

Comment:
This throws away the newlines, corrupting the data passd to `UpdateRemoteSearch`.

Adding `buffer.Append("\n");` after this fixes things.